### ISSUE TITLE
ENH: Use alarms to determine when state files should be written

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#612]
The alarms module will facilitate periodic writing of state files (e.g. once a year or every 10 days).

**Comment by bartnijssen** [comment]
Will it be able to handle things like "first of the month", which would be an irregular delta

**Comment by jhamman** [comment]
Yes, this would be the benefit of using the alarms module rather than the current system. Any frequency that we can write out data (days, months, years, specific date, nsteps, end), we could write out state files.
